### PR TITLE
fix(dave2d): required when using LVGL with LV_COLOR_DEPTH 32

### PR DESCRIPTION
**Pull request by jeremy-baker**
LV_COLOR_FORMAT_RGB888 is a 3 byte colour format, the Dave2D and GLCDC only support 4 byte colour formats for RGB888 and ARGB8888 

A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/draw/renesas/dave2d/lv_draw_dave2d_utils.c [hunk 1]
```diff
@@ -75,7 +75,7 @@ d2_s32 lv_draw_dave2d_cf_fb_get(void)
             d2_fb_mode = d2_mode_argb8888;
             break;
         case DISPLAY_IN_FORMAT_32BITS_RGB888: ///< RGB888,   32 bits
-            d2_fb_mode = d2_mode_rgb888;
+            d2_fb_mode = d2_mode_argb8888;
```
Comment (uLipe) [inline src/draw/renesas/dave2d/lv_draw_dave2d_utils.c]: I think this line deserve a comment or something like this:
```
case DISPLAY_IN_FORMAT_32BITS_ARGB888:
case DISPLAY_IN_FORMAT_32BITS_RGB888:
             d2_fb_mode = d2_mode_argb8888;
```

This way it makes explicitly that only mode supported is 4-byte colour pattern, even if the alpha channel is not used.

Comment (jeremy-baker) [reply src/draw/renesas/dave2d/lv_draw_dave2d_utils.c]: Comment added

**Review by uLipe — APPROVED** [review]

**Review by kisvegabor — COMMENTED** [review]
Looks good, thank you!

**Review by kisvegabor — APPROVED** [review]

**Review by liamHowatt — APPROVED** [review]
Thanks, I assume there's no `d2_mode_xrgb8888`

**Pull request merged**